src/api.ts

```diff
@@ -2,23 +2,30 @@ import { Unleash } from "unleash-client";
 import { Context } from "unleash-client/lib/context";
 import * as clientManager from "./unleash_client_manager";
 import * as contextManager from "./context_manager";
+import { log } from "./logger";
 
 export async function isFeatureEnabled(extensionName: string, featureToggleName: string): Promise<boolean> {
-  if (!extensionName) {
-    throw new Error("Feature toggle extension name can not be empty, null or undefined");
-  }
+  try {
+    if (!extensionName) {
+      throw new Error("Feature toggle extension name can not be empty, null or undefined");
+    }
 
-  if (!featureToggleName) {
-    throw new Error("Feature toggle name can not be empty, null or undefined");
-  }
+    if (!featureToggleName) {
+      throw new Error("Feature toggle name can not be empty, null or undefined");
+    }
 
-  //get unleash client
-  const client: Unleash = await clientManager.getUnleashClient(extensionName);
+    //get unleash client
+    const client: Unleash = await clientManager.getUnleashClient(extensionName);
 
-  // get the context
-  const context: Context = contextManager.getContext(extensionName);
+    // get the context
 
-  //check if the feature is enabled
-  //fallback value is false (3rd parameter)
-  return client.isEnabled(`${extensionName}.${featureToggleName}`, context, false);
+    const context: Context = contextManager.getContext(extensionName);
+
+    //check if the feature is enabled
+    //fallback value is false (3rd parameter)
+    return client.isEnabled(`${extensionName}.${featureToggleName}`, context, false);
+  } catch (e) {
+    log(e);
```

Comment:
what is the exception ? is it clear that this is an authentication issue ?
please also add a message saying "due to authentication error feature toggle is disabled"

Reply:
It catches all exceptions, also errors that where thrown (like missing env variables and etc).
Enables app to run without stopping and dispalying disabled status.